pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -118,146 +127,170 @@ func (r *ReconcileVirtualMachineImport) Reconcile(request reconcile.Request) (re
-					Name:      disk.MustId(), // FIXME:
-					Namespace: "default",     // FIXME:
-					Labels: map[string]string{ // FIXME:
-						"origin": "oVirt",
+		dvs[i] = cdiv1.DataVolume{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "cdi.kubevirt.io/v1alpha1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      diskAttachment.MustId(), // FIXME:
+				Namespace: "default",               // FIXME:
+				Labels: map[string]string{ // FIXME:
+					"origin": "oVirt",
+				},
+			},
+			Spec: cdiv1.DataVolumeSpec{
+				Source: cdiv1.DataVolumeSource{
+					Imageio: &cdiv1.DataVolumeSourceImageIO{
+						URL:           string(secret.Data["apiUrl"]),
+						DiskID:        disk.MustId(),
+						SecretRef:     "ovirt-key", // FIXME, should be created dynamically
+						CertConfigMap: "ovirt-ca",  // FIXME, should be created dynamically
 					},
 				},
-				Spec: cdiv1.DataVolumeSpec{
-					Source: cdiv1.DataVolumeSource{
-						/*
-							Imageio: &cdiv1.DataVolumeSourceImageIO{
-								URL:           secret.Data["apiUrl"],
-								DiskID:        vm.MustDiskAttachments().Slice()[0].MustDisk().MustId(),
-								SecretRef:     "",
-								CertConfigMap: "",
-							},
-						*/
+				// TODO: Would be great to add storageClassName which we should get from the mapping.
+				PVC: &corev1.PersistentVolumeClaimSpec{
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
 					},
-					PVC: &corev1.PersistentVolumeClaimSpec{
-						AccessModes: []corev1.PersistentVolumeAccessMode{
-							corev1.ReadWriteOnce,
-						},
-						Resources: corev1.ResourceRequirements{
-							Requests: corev1.ResourceList{
-								corev1.ResourceStorage: quantity,
-							},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: quantity,
 						},
 					},
 				},
 			},
-		)
+		}
 	}
 	return dvs
 }
 
 // newVMForCR returns a VM specification of the fetched oVirt VM
 func newVMForCR(vm *ovirtsdk.Vm, dvs []cdiv1.DataVolume, con *ovirtsdk.Connection) *kubevirtv1.VirtualMachine {
+	// Labels definition:
 	labels := map[string]string{
 		"origin": "oVirt",
 	}
+	// Volumes definition:
+	volumes := make([]kubevirtv1.Volume, len(dvs))
+	for i, dv := range dvs {
+		volumes[i] = kubevirtv1.Volume{
+			Name: fmt.Sprintf("dv-%v", i),
+			VolumeSource: kubevirtv1.VolumeSource{
+				DataVolume: &kubevirtv1.DataVolumeSource{
+					Name: dv.Name,
+				},
+			},
+		}
+	}
+	// Disks definition:
+	disks := make([]kubevirtv1.Disk, len(dvs))
+	for i := range dvs {
+		disks[i] = kubevirtv1.Disk{
+			Name: fmt.Sprintf("dv-%v", i),
+			DiskDevice: kubevirtv1.DiskDevice{
+				Disk: &kubevirtv1.DiskTarget{
+					Bus: "virtio", // FIXME:
+				},
+			},
+		}
+	}
+	running := false
 	return &kubevirtv1.VirtualMachine{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      vm.MustName(), // FIXME:
 			Namespace: "default",     // FIXME:
 			Labels:    labels,
 		},
 		Spec: kubevirtv1.VirtualMachineSpec{
```

Comment:
We should also take care of the boot sequence.

Reply:
Correct. Ovirt can have only one bootable disk, so I will set this disk booOrder to 0, and I won't set any bootOrder to other disks.